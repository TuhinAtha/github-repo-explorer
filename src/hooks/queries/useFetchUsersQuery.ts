import { useQuery } from "@tanstack/react-query";
import Users from "../../models/Users";
import queryKeys from "./QueryKeys";
import { AxiosError } from "axios";
import GitHubAxiosInstance from "../../GitHubAxiosInstance";

const DEFAULT_PER_PAGE = 5

export type UsersQueryProps = {
  q: string,
  per_page?: number
}

const useFetchUsersQuery = (props: UsersQueryProps) => {
  const { q = '', per_page = DEFAULT_PER_PAGE } = props
  return useQuery<Users, AxiosError>({
    queryKey: [queryKeys.SEARCH_USERS, { ...props }],
    queryFn: async () => {
      const response = await GitHubAxiosInstance.get(`/search/users?q=${encodeURIComponent(props.q)}&per_page=${per_page}`)
      return response.data
    },
    enabled: Boolean(q),
    keepPreviousData: Boolean(q)
  })

}
export default useFetchUsersQuery;
