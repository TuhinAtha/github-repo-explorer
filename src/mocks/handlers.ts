import { rest } from 'msw'

export const handlers = [rest.get(
  process.env.REACT_APP_GITHUB_BASE_URL + '/search/users',
  (req, res, ctx) => {
    return res(
      ctx.json([{
        login: 'username1'
      }])
    )
  }),
rest.get(
  process.env.REACT_APP_GITHUB_BASE_URL + '/users/:username/repos',
  (req, res, ctx) => {
    return res(
      ctx.json([{
        name: 'repo1',
        description: 'repo desctipiton 1',
        stargazers_count: 100
      }])
    )
  }
)]
