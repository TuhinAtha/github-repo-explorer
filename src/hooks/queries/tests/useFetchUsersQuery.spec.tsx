import { renderHook, waitFor } from '@testing-library/react'
import useFetchUsersQuery from '../useFetchUsersQuery'
import { ReactQueryClientProviderWrapper } from '../../../mocks/ReactQueryClientProvider'

describe('Hooks -> useFetchUsersQuery', () => {
  it('Should work fine', async () => {
    const { result } = renderHook(
      () =>
        useFetchUsersQuery({
          q: 'user',
        }),
      { wrapper: ReactQueryClientProviderWrapper }
    )

    await waitFor(() => expect(result.current.isSuccess).toBe(true));
    expect(result.current.data).toMatchSnapshot()
  })
})
