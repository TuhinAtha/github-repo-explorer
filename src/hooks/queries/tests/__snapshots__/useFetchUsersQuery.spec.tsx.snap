// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hooks -> useFetchUsersQuery Should work fine 1`] = `
Array [
  Object {
    "login": "username1",
  },
]
`;
